### ISSUE TITLE
Now updating the server

### ISSUE THREAD

**Issue by ndmitchell** [haskell/vscode-haskell#380]
When starting I get:

```
Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper) Version 0.7.1.0 x86_64 ghc-8.10.3
Current directory: c:\Neil\shake
Operating system: mingw32
Arguments: ["--lsp"]
Cradle directory: c:\Neil\shake
Cradle type: Cabal

Tool versions found on the $PATH
cabal:		3.4.0.0
stack:		2.3.3
ghc:		8.10.3


Consulting the cradle to get project GHC version...
Project GHC version: 8.10.3
haskell-language-server exe candidates: ["haskell-language-server-8.10.3.exe","haskell-language-server-8.10.exe","haskell-language-server.exe"]
Cannot find any haskell-language-server exe, looked for: haskell-language-server-8.10.3.exe, haskell-language-server-8.10.exe, haskell-language-server.exe
[Error - 12:36:56] Connection to server got closed. Server will not be restarted.
```

I expected that the server would download the binary for me. But it didn't. Looking at the config:

![image](https://user-images.githubusercontent.com/1651197/115111559-d007dc00-9f78-11eb-9d7b-3dcee2ba202f.png)

So it seems that it won't auto-download because I don't have "Language Server Variant" set to "haskell-language-server". Indeed, I don't have this in my config, its the default.

I tried changing a bunch of things, like making it explicit in my .json and trying "prompt" behaviour, but nothing seems to allow the addin to download haskell-language-server. How come?

**Comment by jneira** [comment]
Hi Neil, it is a really weird behaviour and i assume it works in the general case, or we would have a ton of issues about. :wink: 
I've observed you have a really old haskell-language-server-wrapper version (0.7.1.0). Maybe you have it in PATH and the extension cant use a newer, automatically downloaded, working version?

**Comment by ndmitchell** [comment]
Indeed, I had `C:\Users\ndmit_000\AppData\Roaming\cabal\bin\haskell-language-server-wrapper.EXE` on my PATH. After deleting that it works. Why is the existence of `C:\Users\ndmit_000\AppData\Roaming\cabal\bin\haskell-language-server-wrapper.EXE` a relevant variable? Is that what we expect developers to do to create it? If we see that and thus decide not to bother downloading (not unreasonable) could we include it in the log message?

**Comment by jneira** [comment]
> Why is the existence of C:\Users\ndmit_000\AppData\Roaming\cabal\bin\haskell-language-server-wrapper.EXE a relevant variable?

It is not relevant per se, the extension prioritize executables in `%PATH%` whatever will be its concrete path. 

>  If we see that and thus decide not to bother downloading (not unreasonable) could we include it in the log message?

I think newer versions of hls-wrapper prints the executable path and it could help to trace the issue. But the extension could tell us that it is gonna use the executable in `%PATH%` and skip the download.
